### PR TITLE
Fixes #30620 - Fix CR creation from the API

### PR DESCRIPTION
**Pull request by shiramax**
Fixes a bug introduced in https://github.com/theforeman/foreman/pull/7814.

### CODE REVIEW

**Review by ezr-ondrej — APPROVED** [review]
I guess we could rather refactor the CRs to have the same way of doing this, but this is good fix to get in fast :+1:

File: app/controllers/api/v2/compute_resources_controller.rb [hunk 1]
```diff
@@ -68,7 +68,11 @@ def change_datacenter_to_uuid(datacenter)
 
       def create
         begin
-          @compute_resource = ComputeResource.new_provider(compute_resource_params.except(:datacenter))
+          if compute_resource_params["provider"] == "Ovirt"
```
Comment (ezr-ondrej) [inline app/controllers/api/v2/compute_resources_controller.rb]: Actually this should compare the `downcase`d version with 'ovirt', just to be sure.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]

**Review by ezr-ondrej — APPROVED** [review]
Great, thanks @shiramax :+1:
WWFJ.

**Comment by ezr-ondrej** [comment]
I think we're seeing the katello errors all over, so I think we are fine (and the errors are not related), but I'd like to be sure.
@tbrisker do you know what's happening?

**Review by tbrisker — APPROVED** [review]
Perhaps we should refactor the conversion to uuid, but let's first fix this bug.

**Comment by tbrisker** [comment]
2.2 - ba93142f0
2.1 - 8b2141b20

**Pull request merged**